src/containers/Login/Login.tsx

```diff
@@ -0,0 +1,35 @@
+import { useParams } from 'react-router-dom';
+import { useAlertBanner } from '~/contexts';
+import { mapLoginError, safeEnv } from '~/helpers';
+import { LoginForm } from './LoginForm';
+
+export type LoginFormProps = {
+  onSuccess: (username: string) => void | Promise<void>;
```

Comment:
`Promise<void>` is redundant here I believe.

Reply:
This allows callers to pass an async function or a non-async function.